tests/5.1/metadirective/test_metadirective_nothing.c

```diff
@@ -0,0 +1,96 @@
+//===--------------------- test_metadirective_nothing.c ---------------------===//
+//
+// OpenMP API Version 5.1 Nov 2020
+// 
+// Test for nothing directive within metadirectives. Runs a variety of
+// metadirectives that check if the nothing directive is properly rendered.
+// Primarily tests based on the fact that no matter what 'when' clause is 
+// rendered it should result in nothing, and thus no additional pragma should
+// be created. Thus, the threads should remain unchanged through this process
+// and the compiler should handle it properly. Handles both device and host
+// based tests.
+//
+////===---------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+int metadirectiveOnDevice() {
+   int errors = 0;
+   int A[N];
+
+   for (int i = 0; i < N; i++) {
+      A[i] = 0;
+   }
+
+   // We expect at least one of these when conditons to eval to true, thus having the nothing directive utilized
+   #pragma omp metadirective \
+      when( device={kind(nohost)}: nothing ) \
+      when( device={arch("nvptx")}: nothing) \
+      when( implementation={vendor(amd)}: nothing ) \
+      default( target parallel for map(tofrom: A) )
+      {
+         for (int i = 0; i < N; i++) {
+            A[i] += omp_in_parallel();
+         }
+      }
+
+   for (int i = 0; i < N; i++) {
+      OMPVV_TEST_AND_SET(errors, A[i] != 0 || A[i] != 1);
+   }
+
+   OMPVV_INFOMSG("Test ran with a number of available devices greater than 0");
+   OMPVV_INFOMSG_IF(A[0] == 0, "Test recognized device was of arch/vendor/kind nvidia, amd, or nohost");
```

Comment:
The test is running from the host...it will ALWAYS go for the default option. Even if Nvidia or AMD devices are available, if they are not on the said device they will always go to default.

Reply:
I was under the impression that the metadirective would query device info regardless of the "region" it was in. It's checking device info and then converting into a respective construct. A very big use case for metadirectives would be determining changes at runtime based on the device being used, or whether to run something on host or parallel depending on the system. If it needed to be in a target region to access that information, then the metadirective feels more redundant since you would need to program a target region and test it on multiple devices, or add preprocessor #if statements to get that dynamic logic.

Reply:
No, it does not work that way. Think of a system that has both AMD and Nvidia GPU devices...what would happened then ? Metadirective checks what "is" and makes the selection. Yes, your point is correct, it will make a check at runtime to check which device is being used and make the appropriate selection. *Being used* here means that the code is already offloaded. Even with the code offloaded you could do different things on an NVIDIA device (say use loop) vs.  AMD ( teams distribute parallel for simd) as NVIDIA has better loop implementation.

Reply:
Gotcha, fair enough. Thank you for the explanation, I'll patch it now.